tests/foreman/cli/test_repository.py

```diff
@@ -75,68 +83,332 @@
 from robottelo.constants.repos import FEDORA27_OSTREE_REPO
 from robottelo.datafactory import invalid_http_credentials
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.datafactory import valid_docker_repository_names
 from robottelo.datafactory import valid_http_credentials
-from robottelo.decorators.host import skip_if_os
 from robottelo.helpers import get_data_file
-from robottelo.host_info import get_host_os_version
-from robottelo.test import CLITestCase
 from robottelo.utils.issue_handlers import is_open
 
 
-class RepositoryTestCase(CLITestCase):
-    """Repository CLI tests."""
+YUM_REPOS = (FAKE_0_YUM_REPO, FAKE_1_YUM_REPO, FAKE_2_YUM_REPO, FAKE_3_YUM_REPO, FAKE_4_YUM_REPO)
+PUPPET_REPOS = (
+    FAKE_1_PUPPET_REPO,
+    FAKE_2_PUPPET_REPO,
+    FAKE_3_PUPPET_REPO,
+    FAKE_4_PUPPET_REPO,
+    FAKE_5_PUPPET_REPO,
+)
+INVALID_CREDS = invalid_http_credentials()
```

Comment:
I appreciate the effort to make these parameters programmatic and consistent, which we should keep.

I would like to avoid calling these functions at module import, as well as the call to `parametrized` at L294. Let's find a strategy to delay these calls to test generation/collection.

Reply:
`invalid_http_credentials()`, `parametrized()`, etc., have to be called during module import by the test method decoration, but I've moved them into each individual test's `@pytest.mark.parametrize()`, because otherwise duplicate repo names causes problems during repo fixture setup, now that the repo cleanup code has been removed.